src/NuGet.Core/NuGet.Build.Tasks/GetGlobalPropertyValueTask.cs

```diff
@@ -0,0 +1,84 @@
+using Newtonsoft.Json;
+using Task = Microsoft.Build.Utilities.Task;
+
+
+namespace NuGet.Build.Tasks
+{
+    public class GetGlobalPropertyValueTask : Task
+    {
+        [Required]
+        public string PropertyName { get; set; }
+
+        [Output]
+        public string GlobalPropertyValue { get; set; }
+
+        [Output]
+        public bool CheckCompleted { get; set; }
+
+        public override bool Execute()
+        {
+            var globalProperties = GetGlobalProperties();
+
+            if (globalProperties != null)
+            {
+                var dictionaryWithOrdinal = globalProperties.ToDictionary(i => i.Key, i => i.Value, StringComparer.OrdinalIgnoreCase);
+                CheckCompleted = true;
+                if (dictionaryWithOrdinal.TryGetValue(PropertyName, out string globalProperty))
+                {
+                    GlobalPropertyValue = globalProperty;
+                }
+            }
+            return !Log.HasLoggedErrors;
+        }
+
+        /// <summary>
+        /// Get the global property from the IBuildEngine API. 
+        /// </summary>
+        /// <returns>Returns the dictionary with the global properties if they can be accessed. <see langword="null"/> otherwise, which means that the msbuild version doesn't implement this API. </returns>
+        internal IReadOnlyDictionary<string, string> GetGlobalProperties()
+        {
+#if IS_CORECLR
+            // MSBuild 16.5 and above has a method to get the global properties, older versions do not
+            IReadOnlyDictionary<string, string> msBuildGlobalProperties = BuildEngine is IBuildEngine6 buildEngine6
+                ? buildEngine6.GetGlobalProperties()
+                : null;
+#else
+            IReadOnlyDictionary<string, string> msBuildGlobalProperties = null;
+
+            // MSBuild 16.5 added a new interface, IBuildEngine6, which has a GetGlobalProperties() method.  However, we compile against
+            // Microsoft.Build.Framework version 4.0 when targeting .NET Framework, so reflection is required since type checking
+            // can't be done at compile time
+            var buildEngine6Type = typeof(IBuildEngine).Assembly.GetType("Microsoft.Build.Framework.IBuildEngine6");
+
+            if (buildEngine6Type != null)
+            {
+                var getGlobalPropertiesMethod = buildEngine6Type.GetMethod("GetGlobalProperties", BindingFlags.Instance | BindingFlags.Public);
+
+                if (getGlobalPropertiesMethod != null)
+                {
+                    try
+                    {
+                        if (getGlobalPropertiesMethod.Invoke(BuildEngine, null) is IReadOnlyDictionary<string, string> globalProperties)
+                        {
+                            msBuildGlobalProperties = globalProperties;
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        // Ignored
+                    }
+                }
```

Comment:
I don't know what NuGet's policy is about log messages, I would normally log a message for the `else` on this `if`-statement, and the `catch` block as well.

It would help my future self when reviewing `.binlog`'s.